src/main/isValidLocalDate.ts

```diff
@@ -0,0 +1,44 @@
+import {
+  registerDecorator,
+  ValidationArguments,
+  ValidationOptions,
+  ValidatorConstraint,
+  ValidatorConstraintInterface
+} from 'class-validator'
+
+import { LocalDate } from './interface/date'
+
+const EXPECTED_YEAR_LENGTH: number = 4
+
+@ValidatorConstraint()
+export class IsValidLocalDateConstraint implements ValidatorConstraintInterface {
+
+  validate (value: any | LocalDate, args: ValidationArguments): boolean {
+    if (value == null) {
+      return true
+    }
+
+    if (!value.toMoment || !value.year) {
+      return false
+    }
+
+    return value.toMoment().isValid() && Number(value.year).toString().length === EXPECTED_YEAR_LENGTH
```

Comment:
Shall we reorder the conditions so that we do not attempt creating moment instance if year is not 4 digits long? Small optimisation I think.